### DISCUSSION TITLE
Does Sionna include a factory scene ?

### DISCUSSION THREAD

**Discussion by SonDinhVan** [NVlabs/sionna#325] [General]
Hi,

I am working on a project where I need to generate channel state information for the indoor factory environment.

I looked at the scenes example in Sionna, however, I didn't find any similar scenarios. Is there anyway to work around with it ?

Also, if I have, let's say 4 APs and 4 UEs, each AP has 16 antennas and each UE has a single antenna, is it practical to generate channel state information for each pair AP - UE using Sionna, considering this process can be repeated more than 100000 times ?

Thanks very much for your kind support. Really appreciate.

Kind regards,

SDinh

**Comment by jhoydis** [comment]
Hi,

We do not have any factory scene, yet. Is there an example of such a scene anywhere available? It should not be difficult to make one.

Essentially you are asking to simulate the channels between 4 APs and 4x100000 UEs. You should be able to place (depending on your GPU memory) around 200 UEs per batch. That means that you would need to run 2000 iterations. Assuming that each batch takes 2s to compute (depending on the number of paths, considered effects), it would take only a bit more than one hour to compute all of the channel impulse responses. So this looks largely feasible.

**Reply by SonDinhVan** [reply]
Good morning,

Many thanks for your quick response. 

I wonder which scene might be suitable for the rendering of Sionna?

If I want to create one, is there any tutorial/example to look at?

When it comes to channel generation, more than an hour seems very feasible to me. Does this process sound correct to you ? Note that in my case, the UE is not random, but they move along a path, i.e., their location changes in the time domain.

1. Create a list of 100000 UEs, and adjust the location of each UE to match the path.
2. The next step is what you mentioned above.

Finally, has Sionna been optimized on Mac M1 yet? 

Thank you again, and I hope you have a wonderful weekend.

**Reply by jhoydis** [reply]
Yes, sounds good. No, Sionna has not been optimised for M1 yet.
